driver-core/src/main/com/mongodb/connection/netty/ReadTimeoutHandler.java

```diff
@@ -40,13 +40,16 @@ public ReadTimeoutHandler(final long readTimeout) {
     }
 
     void scheduleTimeout(final ChannelHandlerContext ctx) {
-        timeout = ctx.executor().schedule(new ReadTimeoutTask(ctx), readTimeout, TimeUnit.MILLISECONDS);
+        if (timeout == null) {
+            timeout = ctx.executor().schedule(new ReadTimeoutTask(ctx), readTimeout, TimeUnit.MILLISECONDS);
+        }
     }
 
     void removeTimeout(final ChannelHandlerContext ctx) {
         if (ctx.channel().eventLoop().inEventLoop()) {
             if (timeout != null) {
```

Comment:
Changed this to an assertion, to ensure its called from the event loop.

The actual calling is done via `NettyStream`